src/components/TableColumnHeader.tsx

```diff
@@ -0,0 +1,30 @@
+import { ArrowUpDown } from 'lucide-react';
+import { Button } from './ui/button';
+import { Column } from '@tanstack/react-table';
+
+type TableColumnHeaderProps<TData, TValue> = {
+  column: Column<TData, TValue>;
+  title: string;
```

Comment:
You don't need the whole column in this component. Instead you should pass a sorting handler function. `() => column.toggleSorting(column.getIsSorted() === 'asc')` should be defined in a parent and just passed here.

The title should be `ReactNode` type and be passed as `children`